### PR TITLE
feat: add fennel-ls support

### PR DESCRIPTION
**Pull request by gh-liu**
Untested on windows.

### CODE REVIEW

**Comment by glepnir** [comment]
does this server not support windows?

File: lua/lspconfig/server_configurations/fennel-ls.lua [hunk 1]
```diff
@@ -0,0 +1,17 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    cmd = { "fennel-ls" },
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/fennel-ls.lua:5]:
```suggestion
    cmd = { 'fennel-ls' },
```

File: lua/lspconfig/server_configurations/fennel-ls.lua [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    filetypes = { "fennel" },
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/fennel-ls.lua:6]:
```suggestion
    filetypes = { 'fennel' },
```

File: lua/lspconfig/server_configurations/fennel-ls.lua [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    root_dir = function(dir) return util.find_git_ancestor(dir) end,
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/fennel-ls.lua:7]:
```suggestion
    root_dir = function(dir)
         return util.find_git_ancestor(dir)
    end,
```

File: lua/lspconfig/server_configurations/fennel-ls.lua [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    settings = {}
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/fennel-ls.lua:8]:
```suggestion
    settings = {},
```

**Comment by gh-liu** [comment]
> does this server not support windows?



> does this server not support windows?


I guess it's possible, but I haven't tried it yet.

**Comment by glepnir** [comment]
so we can simply add the server cmd in windows. if it not works fine we can fix it according the error message

```lua

local cmd = 'fennel-ls'
if vim.fn.has('win32') then
  cmd = { 'cmd.exe', '/C', 'fennel-ls'}
end

```

**Comment by gh-liu** [comment]
> 

I have tried in wsl2, but it's not work. 
The error in lsp.log is  `.../vim/lsp/rpc.lua:732  "rpc" "cmd.exe" "stderr"  "'fennel-ls' is not recognized as an internal or external command,\r\noperable program or batch file.\r\n" ` , but I do not known how to compile is to a windows EXE file.

**Comment by glepnir** [comment]
hmmm okay then just need fix the commit message.

**Comment by gh-liu** [comment]
> hmmm okay then just need fix the commit message.

ok

**Pull request merged**